### PR TITLE
Update NLU tests

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- remove duplicated tests
- reduce number of train epochs in tests
- do not train model if not needed

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tabergma** [comment]
@tmbo Reduced the training time from ~45 min to ~35 min. Is that fine for now? Please keep in mind that model training of certain models takes longer now as we switched to sequential models. Also we added quite a few tests as the NLU components were just poorly tested before.

File: examples/moodbot/config.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 language: en
 
-pipeline: "pretrained_embeddings_spacy"
+pipeline:
+  - name: "SpacyNLP"
+  - name: "SpacyTokenizer"
+  - name: "SpacyFeaturizer"
+  - name: "RegexFeaturizer"
```
Comment (Ghostvv) [inline examples/moodbot/config.yml]: lexico...? with pos features?

File: examples/knowledgebasebot/config.yml [hunk 2]
```diff
@@ -1,5 +1,15 @@
 language: en
-pipeline: supervised_embeddings
+
+pipeline:
+  - name: "WhitespaceTokenizer"
+  - name: "RegexFeaturizer"
```
Comment (Ghostvv) [inline examples/knowledgebasebot/config.yml]: lexico...?

File: examples/concertbot/config.yml [hunk 3]
```diff
@@ -1,6 +1,15 @@
 language: en
 
-pipeline: supervised_embeddings
+pipeline:
+  - name: "WhitespaceTokenizer"
+  - name: "RegexFeaturizer"
```
Comment (Ghostvv) [inline examples/concertbot/config.yml]: lexico...?

File: tests/nlu/test_config.py [hunk 4]
```diff
@@ -99,43 +101,78 @@ def test_default_config_file():
     assert len(final_config) > 1
 
 
-def test_set_attr_on_component(pretrained_embeddings_spacy_config):
-    idx_classifier = pretrained_embeddings_spacy_config.component_names.index(
-        "SklearnIntentClassifier"
-    )
-    idx_tokenizer = pretrained_embeddings_spacy_config.component_names.index(
-        "SpacyTokenizer"
+def test_set_attr_on_component():
```
Comment (Ghostvv) [inline tests/nlu/test_config.py]: should we test set attribute on different component? since we plan to deprecate sklearn

File: tests/nlu/test_config.py [hunk 5]
```diff
@@ -99,43 +101,78 @@ def test_default_config_file():
...
+
+    assert _config.for_component(idx_tokenizer) == {"name": "SpacyTokenizer"}
+    assert _config.for_component(idx_classifier) == {
         "name": "SklearnIntentClassifier",
         "C": 324,
     }
 
 
-def test_override_defaults_supervised_embeddings_pipeline(supervised_embeddings_config):
+def test_override_defaults_supervised_embeddings_pipeline():
```
Comment (Ghostvv) [inline tests/nlu/test_config.py]: should we test override on different component? since we plan to deprecate sklearn

File: examples/moodbot/config.yml (hunk 1 above)
Comment (tabergma) [reply examples/moodbot/config.yml]: good point, will add it
